src/main/java/com/powsybl/network/hypothesis/GeneratorsStartupAlgorithm.java

```diff
@@ -0,0 +1,274 @@
+                    .startupType(StartupType.EMPIL_ECO)
+                    .canStart(true)
+                    .countries(new ArrayList<>())
+                    .startupGroups(startupGroupsPerConnectedComponent.get(component))
+                    .startedGroups(new ArrayList<>())
+                    .imposedPower(0)
+                    .build();
+
+            // log component num
+            LOGGER.debug("Component: {}", component.getNum());
+
+            evaluateConsumption(network, startupZone); // real consumption + losses
+
+            if (startupZone.getStartupType() == StartupType.EMPIL_ECO) {
+                economicalStacking(startupZone);
+            }
+
+            if (startupZone.isCanStart()) {
+                final double[] startedProduction = {0};
+
+                startupZone.getStartedGroups().forEach(startupGroup -> {
+                    startupGroup.getGenerator().setTargetP(startupGroup.getSetPointPower());
+                    startedProduction[0] += startupGroup.getSetPointPower();
+
+                    if ((startupGroup.getGenerator().getTargetP() >= pThreshold) &&
+                            (startupGroup.getGenerator().getReactiveLimits().getMaxQ(0) - startupGroup.getGenerator().getReactiveLimits().getMinQ(0) >= qThreshold)) {
+                        startupGroup.getGenerator().setVoltageRegulatorOn(true);
+                        LOGGER.info("Primary Voltage Control in operation");
+                    }
+                });
+
+                if (Math.abs(startedProduction[0] - startupZone.getConsumption()) > 1)  {
+                    LOGGER.error("Wrong starting production units: load + loss = {} MW; started production {} MW", startupZone.getConsumption(), startedProduction);
+                }
+            }
+        });
+    }
+
+    private Map<Component, List<StartupGroup>> prepareStartupGroupsPerConnectedComponentMap(Network network) {
+        Map<Component, List<StartupGroup>> startupGroupsPerConnectedComponent = new HashMap<>();
+        network.getGeneratorStream().forEach(generator -> {
+            StartupGroup startupGroup = new StartupGroup(false, 0, 0, false, generator);
+            Component component = generator.getRegulatingTerminal().getBusBreakerView().getConnectableBus().getConnectedComponent();
+            startupGroupsPerConnectedComponent.computeIfAbsent(component, k -> new ArrayList<>());
+            List<StartupGroup> l = startupGroupsPerConnectedComponent.get(component);
+            l.add(startupGroup);
+        });
+        return startupGroupsPerConnectedComponent;
+    }
+
+    // calculate  zone global consumption + losses
+    private void evaluateConsumption(Network network, StartupZone startupZone) {
+        final double[] consZone = {0};
+        final double[] consFictitious = {0};
+
+        network.getLoadStream().forEach(load -> {
+            if (load.getTerminal().getBusView().getBus().getConnectedComponent().getNum() == startupZone.getNum()) {
```

Comment:
Maybe check that `getBus` does not return `null` first or you will have a `NullPointerException`